### ISSUE TITLE
Question about mulitple forward and gradient accumulation

### ISSUE THREAD

**Issue by slyviacassell** [salesforce/LAVIS#82]
I have seen that some models call `forward` multiple times. And as far as I know, calling `forward` multiple times is equal to gradient accumulation.
For instance, in ALBEF model, the `self.text_encoder.bert` is called three times out of context `with torch.no_grad()`. Hence, the gradient of it is accumulated. Would you mind providing the results without calling it multiple times for comparison?

**Comment by dxli94** [comment]
Hi @slyviacassell ,

Yes. This is the expected behavior of ALBEF. For MLM, only the first 6 Bert layers are involved in the forward pass. While for ITM, the first 6 layers for forwarding text embeddings, the last 6 layers for multimodal embeddings.

Thanks.

**Comment by dxli94** [comment]
Hi @slyviacassell, yes, you are right. Both ITM and MLM involves the full forward pass of text encoder. I've updated my previous comment - since it was incorrect. Thanks.
